lib/Doctrine/DBAL/Platforms/DB2Platform.php

```diff
@@ -878,4 +881,57 @@ protected function getReservedKeywordsClass()
     {
         return Keywords\DB2Keywords::class;
     }
+    
+    /**
+     * Prepare ORDER BY string for OVER() if applicable
+     * 
+     * @param  string $query
+     * 
+     * @return string
+     */
+    private function getOrderByForOver($query)
+    {
+        //preset empty array
+        $orderByArray = [];
+
+        // remove extra white space
+        $query = preg_replace('/\s+/', ' ', $query);
+
+        //determine if 'ORDER BY' is part of the query
+        $orderByFound = preg_match('/(\bORDER\b)(\s*)(\bBY\b)(?!.*\b\1\b)/i', $query, $orderByPositionArray, PREG_OFFSET_CAPTURE);
+
+        // ORDER BY found in query string
+        if (0 !== $orderByFound) {
+            $orderByPosition = $orderByPositionArray[0][1];
+
+            $queryArray = preg_split('/[, ]/', substr($query, 0, $orderByPosition -1));
+            $orderByArray = explode(',', substr($query, $orderByPosition + strlen('ORDER BY')));
+
+            foreach ($orderByArray as $orderIndex => $orderValue) {
+                $splitOrder = explode(' ', $orderValue);
+
+                foreach ($splitOrder as $splitIndex => $splitValue) {
+                    switch (strtoupper($splitValue)) {
+                        case 'ASC':
+                        case 'DESC':
+                            break;
+                        default:
+                            $splitOrder[$splitIndex] = array_search($splitValue, $queryArray) === false
+                                                       ? $splitValue
+                                                       : $queryArray[array_search($splitValue, $queryArray)+2];
```

Comment:
Could you pls add some tests explaining this logic? As I read it, the code tries to find the same column in `SELECT` and if fails, uses the expression as is, otherwise, uses something else (the `+ 2` part). The variable names `$splitOrder` (which is an array of strings) and `$splitValue` look confusing. Is it possible to rename it using some SQL terms?

Not sure if this code properly handles the case when a selected column has an alias (with or without the `AS` keyword) like `SELECT foo foo_alias, bar AS bar_alias`. Not sure what the expected result in this case is. This should be covered by a test too.

Reply:
The logic... DB2 (on IBMi) is kinda strict on certain statements...
The reason for that, and I still not to address the possibility of subqueries here, is -> by default, if I use the entity and slap a limit/offset on it, it always adds a "as xxx" to it. within the "OVER()" you have to sort by the "AS" declaration otherwise DB2 can't handle it.
However!, I have to see how IBM reacts to your statement; without the "AS"...
Dang... I thought that would be an easy ride, like he persistent connection I added to ZF2 for IBMi... LOL